### PR TITLE
Add rocrand, rocsparse, and rocthrust externals to Crusher

### PR DESCRIPTION
**Pull request by jrood-nrel**

### CODE REVIEW

**Review by PaulMullowney — COMMENTED** [review]
Did this build for you? I think we also need a block for rocprim and we need a small change to spack/.../hypre/package.py

**Comment by jrood-nrel** [comment]
I haven't tried building nalu-wind with `+rocm`. Just assuming this would help.

**Comment by PaulMullowney** [comment]
> I haven't tried building nalu-wind with `+rocm`. Just assuming this would help.

Ok. Don't merge this yet. I don't think its complete.

**Comment by PaulMullowney** [comment]
> > I haven't tried building nalu-wind with `+rocm`. Just assuming this would help.
> 
> Ok. Don't merge this yet. I don't think its complete.

We need to take /hip off the end of the prefix below. The reason for this is a recent change to the hip spack package.py
  hip:
    version: [5.2.0]
    buildable: false
    externals:
      - spec: hip@5.2.0
        prefix: /opt/rocm-5.2.0/hip
        modules:
          - rocm/5.2.0
          - craype-accel-amd-gfx90a

File: configs/crusher/packages.yaml [hunk 1]
```diff
@@ -8,12 +8,48 @@ packages:
...
+        prefix: /opt/rocm-5.2.0/rocrand
+        modules:
+          - rocm/5.2.0
+          - craype-accel-amd-gfx90a
+  rocthrust:
+    version: [5.2.0]
+    buildable: false
+    externals:
+      - spec: rocthrust@5.2.0
+        prefix: /opt/rocm-5.2.0/thrust
```
Comment (PaulMullowney) [inline configs/crusher/packages.yaml]: Should be
/opt/rocm-5.2.0/rocthrust

**Review by PaulMullowney — APPROVED** [review]

**Review by psakievich — APPROVED** [review]

**Pull request merged**